### PR TITLE
Restructure redux code

### PR DESCRIPTION
**Pull request by mrdrogdrog**
This PR restructures the redux state code. It also moves some types to other directories.

### CODE REVIEW

File: src/api/frontend-config/types.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export interface FrontendConfig {
+  backendUrl: string,
```
Comment (ErikMichelson) [inline src/api/frontend-config/types.ts:2]: There's a tailing comma although no element is following.

Comment (mrdrogdrog) [reply src/api/frontend-config/types.ts]: removed

File: src/components/landing/pages/intro/cover-buttons/cover-buttons.tsx [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
+import { Link } from 'react-router-dom'
 import { ApplicationState } from '../../../../../redux'
 import './cover-buttons.scss'
 
 export const CoverButtons: React.FC = () => {
   useTranslation()
   const user = useSelector((state: ApplicationState) => state.user)
 
-  if (user.status === LoginStatus.ok) {
+  if (!user) {
```
Comment (ErikMichelson) [inline src/components/landing/pages/intro/cover-buttons/cover-buttons.tsx:13]: The logic is inverted here
```suggestion
  if (user) {
```

Comment (mrdrogdrog) [reply src/components/landing/pages/intro/cover-buttons/cover-buttons.tsx:13]: good point

**Pull request merged**